jobclient/src/main/java/com/twosigma/cook/jobclient/Constraint.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) Two Sigma Open Source, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.cook.jobclient;
+
+import com.google.common.base.Preconditions;
+import org.json.JSONArray;
+
+/**
+ * A simple class to represent a constraint.
+ *
+ * A constraint has 3 fields: an attribute, an operator, and a value.
+ * An example of constraints could be: "host","EQUALS", "foo.bar.com".
+ *
+ * For now, only "EQUALS" operator is supported by Cook. One could construct
+ * a constraint with an "EQUALS" operator by
+ *
+ * <pre>
+ *  Constraint c = Constraint.EQUALS.apply("host", "foo.bar.com");
```

Comment:
Nitpick: Can you add the other way that a constraint can be constructed : 

```java
Constraint c = new Constraint("host", Constraint.EQUALS, "foo.bar.com")
```

I find this version easier to read.

Also, I haven't coded java recently so I expect I am wrong but should this be `Constraint.Operator.EQUALS`?

Reply:
I explicitly added the `EQUALS` in the constraint level.

Reply:
Never mind, I see you are binding it at the bottom. Adding a comment to that line